cleanlab/experimental/datalab/datalab.py

```diff
@@ -0,0 +1,385 @@
+        """Set necessary configuration for each IssueManager in a dictionary.
+
+        While each IssueManager defines default values for its arguments,
+        the Datalab class needs to organize the calls to each IssueManager
+        with different arguments, some of which may be
+
+        Parameters
+        ----------
+        issue_types :
+            Dictionary of issue types and argument configuration for their respective IssueManagers.
+            If None, then the `required_defaults_dict` is used.
+
+        required_defaults_dict :
+            Dictionary of default parameter configuration for each issue type.
+
+        Returns
+        -------
+        issue_types_copy :
+            Dictionary of issue types and their parameter configuration.
+            The input `issue_types` is copied and updated with the necessary default values.
+        """
+        if issue_types is not None:
+            issue_types_copy = issue_types.copy()
+            self._check_missing_args(required_defaults_dict, issue_types_copy)
+        else:
+            issue_types_copy = required_defaults_dict.copy()
+        # Check that all required arguments are provided.
+        self._validate_issue_types_dict(issue_types_copy, required_defaults_dict)
+
+        # Remove None values from argument list, rely on default values in IssueManager
+        for key, value in issue_types_copy.items():
+            issue_types_copy[key] = {k: v for k, v in value.items() if v is not None}
+        return issue_types_copy
+
+    @staticmethod
+    def _check_missing_args(required_defaults_dict, issue_types):
+        for key, issue_type_value in issue_types.items():
+            missing_args = set(required_defaults_dict.get(key, {})) - set(issue_type_value.keys())
+            # Impute missing arguments with default values.
+            missing_dict = {
+                missing_arg: required_defaults_dict[key][missing_arg]
+                for missing_arg in missing_args
+            }
+            issue_types[key].update(missing_dict)
+
+    @staticmethod
+    def _validate_issue_types_dict(
+        issue_types: Dict[str, Any], required_defaults_dict: Dict[str, Any]
+    ) -> None:
+        missing_required_args_dict = {}
+        for issue_name, required_args in required_defaults_dict.items():
+            if issue_name in issue_types:
+                missing_args = set(required_args.keys()) - set(issue_types[issue_name].keys())
+                if missing_args:
+                    missing_required_args_dict[issue_name] = missing_args
+        if any(missing_required_args_dict.values()):
+            error_message = ""
+            for issue_name, missing_required_args in missing_required_args_dict.items():
+                error_message += f"Required argument {missing_required_args} for issue type {issue_name} was not provided.\n"
+            raise ValueError(error_message)
+
+    def _get_report(self, k: int, verbosity: int) -> str:
```

Comment:
I think the report still needs some work to be prettier.

V1 of report improvements:

- you are missing the dataset-level issue summary at the top of the report, ie. first part of report should print: `lab.issue_summary` preceded by a sentence explaining what this is.
- each section header currently just says "ISSUE_TYPE" with no other context but it should instead say:
"ISSUE_TYPE issues". 
- should be some space between the name of issue and line dividers, eg: "---- outlier issues -----" instead of "----outlier------"
- these sections should be sorted by the overall severity/prevalence of this issue-type according to `lab.issue_summary`.
- underneath each section header, instead of just "Score: 0.9071" (which does not make sense without explanation), I would say:
"Number of examples with this issue: XYZ  Overall quality in terms of this issue: XYZ" where the latter is your Score.  And when both of these are not available, just show whichever one is available. 
- the printing of `lab.issues.head()` needs to be preceded by a sentence explaining what this means, eg. "Here are the examples in dataset representing the most severe instances of this issue:" 
- I think the default amount of info being shown is a little too much, it should be lower verbosity by default. Eg. a novice user definitely doesn't need to see `weighted_knn_graph` or `radius` or `distance_to_nearest_neighbor` or `confident_joint` etc.
- Instead of just "Info:" I would say something like "Additional Information:"

Reply:
V2 of report improvements:
- synchronization with imagelab report formatting to combine the best aspects of both
- code should detect if this is running inside Jupyter notebook and if so, output a nicer formatted version of the report. In particular, use of pull-downs to reveal more details about each issue.

Reply:
The report is arguably the most important part of this whole class so worth spending the most time on

Reply:
> V2 of report improvements:
> 
> * synchronization with imagelab report formatting to combine the best aspects of both
> * code should detect if this is running inside Jupyter notebook and if so, output a nicer formatted version of the report. In particular, use of pull-downs to reveal more details about each issue.

Tracked in [issue on fork](https://github.com/elisno/cleanlab/issues/10).